crates/matrix-sdk/src/notification_settings/rules.rs

```diff
@@ -210,6 +209,44 @@ impl Rules {
         }
     }
 
+    /// Insert a new rule and optionally delete other custom rules for a room
+    pub(crate) fn insert_new_rule(
+        &mut self,
+        kind: RuleKind,
+        room_id: &RoomId,
+        notify: bool,
+        delete_other_custom_rules: bool,
```

Comment:
It's not clear to me why it's this function's responsibility to delete other custom rules: it has nothing to do with the name, and it seems the caller has the possibility of calling another method, `delete_rules`, to do just that. Could we let the caller handle that, instead? It would make for a smaller API surface, fewer things to test, etc. 

(The caller can even do better: 1. first remove all the custom rules, if needs be, 2. insert the new rule afterwards; that would *not* require the exception in the `delete_rules` call)